src/main/java/messenger/MessengerDatabase.java

```diff
@@ -0,0 +1,43 @@
+package messenger;
+
+import notification.Notification;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MessengerDatabase {
+
+    private HashMap<String, String> idToNameMap;
+
+    private HashSet<Notification> storedNotifications;
+
+    public MessengerDatabase() {
+        storedNotifications = new HashSet<>();
+        try {
+            FileInputStream fis = new FileInputStream("src/main/resources/idToName.ser");
```

Comment:
What should happen when I use MessengerDatabase and there is no such file? Cause right now, when file is absent, you catch exception and progress further adding "Unknown" as senderName and receiverName in preprocessing at 21 line at `MessengerRequestProcessor`

Reply:
Who knows, no one knows.